include/moveit_middleware_benchmark/scenarios/scenario_perception_pipeline.hpp

```diff
@@ -0,0 +1,163 @@
+const std::string PACKAGE_SHARE_DIRECTORY = ament_index_cpp::get_package_share_directory("moveit_middleware_benchmark");
+const std::string TEST_CASES_YAML_FILE =
+    PACKAGE_SHARE_DIRECTORY + "/config/scenario_perception_pipeline_test_cases.yaml";
+}  // namespace
+
+namespace moveit
+{
+namespace middleware_benchmark
+{
+
+using moveit::planning_interface::MoveGroupInterface;
+
+class ScenarioPerceptionPipeline
+{
+public:
+  /** \brief Constructor for interface to send goal poses
+   *  \param [in] move_group_interface_ptr The move_group_interface for sending goal poses
+   */
+  ScenarioPerceptionPipeline(std::shared_ptr<MoveGroupInterface> move_group_interface_ptr);
+
+  /** \brief Given a \e pose_list, sends poses in pose_list to plan,
+   *  if the pipeline in \e sendTargetPose returns true,
+   *  then success_number is increased.
+   *  If the pipeline in \e sendTargetPose returns false,
+   *  failure_number is increased.
+   *  \param [in] test_case The pose list to benchmark
+   *  \return success_number and failure_number in tuple respectively
+   */
+  std::tuple<int, int> runTestCase(const nav_msgs::msg::Path& pose_list);
+
+  /** \brief Given a \e target_pose, sends this pose in order to plan via move_group_interface,
+   *  if the planning is successful,
+   *  planning message is sent to execute trajectory. Then it is returned true.
+   *  If the planning is failed, the execution is ignored and then it is returned false
+   *  \param [in] target_pose The pose to plan
+   *  \return if the planning successful, returns true, or false
+   */
+  bool sendTargetPose(const geometry_msgs::msg::Pose& target_pose);
+
+private:
+  /** move_group_interface_ptr to communicate with move_grouo_server */
+  std::shared_ptr<MoveGroupInterface> move_group_interface_ptr_;
+};
+
+class ScenarioPerceptionPipelineFixture : public benchmark::Fixture
+{
+protected:
+  /* ros node shared ptr*/
+  rclcpp::Node::SharedPtr node_;
+
+  /* move_group_interface_ptr for communicating with move_group_server */
+  std::shared_ptr<MoveGroupInterface> move_group_interface_ptr_;
+
+  /* selected test case index for benchmarking */
+  size_t selected_test_case_index_;
+
+  /** the list of test cases to be tested */
+  std::vector<nav_msgs::msg::Path> test_cases_ = {};
+
+public:
+  ScenarioPerceptionPipelineFixture();
+
+  /** \brief This method runs once each benchmark starts
+   *  \param [in] state
+   *  \return nothing is returned
+   */
+  void SetUp(::benchmark::State& /*state*/);
+
+  /** \brief This method runs as soon as each benchmark finishes
+   *  \param [in] state
+   *  \return nothing is returned
+   */
+  void TearDown(::benchmark::State& /*state*/);
+
+  /** \brief Creates test_cases by filling \e test_cases_
+   *  If the planning is failed, the execution is ignored and then it is returned false
+   *  \return nothing is returned
+   */
+  void createTestCases();
+
+  /** \brief Given a \e test_case_index, returns relevant test case.
```

Comment:
What's the selection criteria?

Reply:
I'm passing arguments to benchmark node like below. User selects one of the test cases and only selected test case is benchmarked.

https://github.com/CihatAltiparmak/moveit_middleware_benchmark/blob/caa54960dd6febab1c82f7b63b93a38616339cda/launch/scenario_perception_pipeline_benchmark.launch.py#L128-L134